tests/e2e/test_read.py

```diff
@@ -0,0 +1,71 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import random
+import string
+import pytest
+
+
+from azure.identity import DefaultAzureCredential
+
+from azstoragetorch.io import BlobIO
+
+from azure.storage.blob import BlobServiceClient
+
+
+@pytest.fixture
+def account_url():
+    account_name = os.environ.get('AZSTORAGETORCH_STORAGE_ACCOUNT_NAME')
+    if account_name is None:
+        raise ValueError(
+            f'"AZSTORAGETORCH_STORAGE_ACCOUNT_NAME" environment variable must be set to run end to end tests.'
+        )
+    return f"https://{account_name}.blob.core.windows.net"
+
+
+@pytest.fixture
+def sample_data():
+    return os.urandom(10)
+
+
+@pytest.fixture
+def container_client(account_url):
+    blob_service_client = BlobServiceClient(
+        account_url, credential=DefaultAzureCredential()
+    )
+    container_name = random_resource_name(8)
+    container = blob_service_client.create_container(name=container_name)
+    yield container
+    container.delete_container()
+
+
+@pytest.fixture
+def blob_client(container_client, sample_data):
+    blob_name = random_resource_name(8)
+    blob_client = container_client.get_blob_client(blob=blob_name)
+    blob_client.upload_blob(sample_data)
+    return blob_client
+
+
+@pytest.fixture
+def blob_url(account_url, container_client, blob_client):
+    return f"{account_url}/{container_client.container_name}/{blob_client.blob_name}"
+
+
+@pytest.fixture
+def blob_io(blob_url):
+    yield BlobIO(
+        blob_url=blob_url, mode="rb"
+    )
+
+def random_resource_name(byte_num):
```

Comment:
For the `random_resource_name()` function, we can probably rename the parameter to `name_length` and then make it's default `8` (e.g. `name_length=8`) so we do not need to specify the length every time we use it.